ejb3/src/main/java/org/jboss/as/ejb3/tx/CMTTxInterceptor.java

```diff
@@ -79,7 +79,7 @@
      */
     protected void endTransaction(final TransactionManager tm, final Transaction tx) {
         try {
-            if (tx != tm.getTransaction()) {
+            if (! tx.equals(tm.getTransaction())) {
```

Comment:
If we throw the wrongTxOnThread error, I would expect that to be an internal (e.g. app server coding bug) issue but am wondering if we should also ensure that the transaction returned from tm.getTransaction() is disassociated from the current thread, so the next use of the current thread, will not see a leaked transaction.  

If you explain more about the use case that led to this change, that might help us understand if further changes are needed here to ensure that the transaction doesn't stay associated with the thread.